docs/external-database.md

```diff
@@ -0,0 +1,144 @@
+# Using an external database
+
+You can use any external Postgres database instead of a local database deployment.
+Tekton Results gets its database configuration either from a Kubernetes Secret
+named `tekton-results-postgres` or from a ConfigMap with the same name.
+
+Tekton Results expects following database configurations to be provided.
+
+| Name | Secret | ConfigMap | Comments |
+| --- | --- | --- | --- |
+| Database User | `POSTGRES_USER` | `DB_USER` | default: postgres |
+| Database Password | `POSTGRES_PASSWORD` | `DB_PASSWORD` | |
+| Database Name | | `DB_NAME` | tekton-results |
+| Database Host | | `DB_HOST` |  default for local: tekton-results-postgres-service.tekton-pipelines.svc.cluster.local |
+
+## Create patches
+
+By default, Tekton Results ships with a local database installation config in the
+release manifest. You can remove them using patches. You may create separate
+patch files or directly add them to kustomization.yaml. Here are the changes:
+
+- Remove database StatefulSet
+
+```yaml
+# delete-database-statefulset.yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: tekton-results-postgres
+  namespace: tekton-pipelines
+$patch: delete
+```
+
+- Remove database Service
+
+```yaml
+# delete-database-service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: tekton-results-postgres-service
+  namespace: tekton-pipelines
+$patch: delete
+```
+
+- Remove database ConfigMap
+
+```yaml
+# delete-datbase-configmap.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: tekton-results-postgres
+  namespace: tekton-pipelines
+$patch: delete
+```
+
+If you want to keep the patches in different files. Create the files as directed
+above and add them to `kustomization.yaml` instead of patches:
+
+```yaml
+patches:
+  - path: delete-database-statefulset.yaml
+  - path: delete-database-service.yaml
+  - path: delete-database-configmap.yaml
+```
+
+## Modifying DB_HOST and DB_NAME
```

Comment:
Don't we need another section describing how to set the remote database username and password?

Reply:
Ah yes, I missed it. I will add it.